crates/biome_js_analyze/src/lint/nursery/no_setup_props_reactivity_loss.rs

```diff
@@ -0,0 +1,386 @@
+use crate::services::semantic::Semantic;
+use biome_analyze::{Rule, RuleDiagnostic, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_js_syntax::{
+    AnyJsAssignmentPattern, AnyJsExpression, AnyJsStatement, JsAssignmentExpression,
+    JsCallExpression, JsExportDefaultExpressionClause, JsMethodObjectMember, JsObjectExpression,
+    JsVariableDeclaration,
+};
+use biome_rowan::{AstNode, AstNodeList, AstSeparatedList, TextRange};
+use biome_rule_options::no_setup_props_reactivity_loss::NoSetupPropsReactivityLossOptions;
+
+declare_lint_rule! {
+    /// Disallow usages that lose the reactivity of `props` passed to `setup`.
+    ///
+    /// Vue's Composition API requires that props passed to the `setup` function
+    /// maintain reactivity. Destructuring props or using member expressions on props
+    /// in the root scope of `setup` will cause the values to lose reactivity.
+    ///
+    /// This rule reports:
+    /// - Direct destructuring of props in setup function parameters
+    /// - Destructuring assignment of props in the root scope of setup
+    /// - Member access patterns that could break reactivity
+    ///
+    /// Note: Destructuring is allowed inside nested functions, callbacks, and
+    /// returned render functions where the reactive context is preserved.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// // Destructuring in setup parameters
+    /// export default {
+    ///   setup({ count }) {
+    ///     // count is no longer reactive
+    ///     return () => h('div', count)
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// // Destructuring in setup root scope
+    /// export default {
+    ///   setup(props) {
+    ///     const { count } = props
+    ///     // count is no longer reactive
+    ///     return () => h('div', count)
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// // Keep props reactive
+    /// export default {
+    ///   setup(props) {
+    ///     watch(() => props.count, () => {
+    ///       console.log(props.count)
+    ///     })
+    ///     return () => h('div', props.count)
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// // Destructuring inside callbacks is OK
+    /// export default {
+    ///   setup(props) {
+    ///     watch(() => props.count, () => {
+    ///       const { count } = props // OK inside callback
+    ///       console.log(count)
+    ///     })
+    ///     return () => {
+    ///       const { count } = props // OK inside render function
+    ///       return h('div', count)
+    ///     }
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub NoSetupPropsReactivityLoss {
+        version: "next",
+        name: "noSetupPropsReactivityLoss",
```

Comment:
Please refer to the contribution guide for the correct name of rule https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#naming-conventions-for-rules

Reply:
This is my mistake, in the linked issue the name `noSetupPropsReactivityLoss` is exactly what I suggested, and I forgot to updated it to reflect current guidelines.

The rule name should be `noVueSetupPropsReactivityLoss`.